commands/createfixpullrequests.go

```diff
@@ -95,113 +104,186 @@ func (cfp *CreateFixPullRequestsCmd) scan(scanSetup *utils.ScanDetails, currentW
+	}
 
-	err = gitManager.Clone(wd, branch)
+	log.Info("Running git add all and commit")
+	commitString := fmt.Sprintf("[🐸 Frogbot] Upgrade %s to %s", impactedPackage, fixVersionInfo.fixVersion)
+	err = cfp.gitManager.AddAllAndCommit(commitString)
 	if err != nil {
 		return err
 	}
 
-	// 'CD' into the temp working directory
-	restoreDir, err := utils.Chdir(wd)
+	log.Info("Pushing fix branch:", fixBranchName)
+	err = cfp.gitManager.Push()
+	if err != nil {
+		return err
+	}
+
+	log.Info("Creating Pull Request form:", fixBranchName, " to:", cfp.details.Branch)
+	prBody := commitString + "\n\n" + utils.WhatIsFrogbotMd
+	return cfp.details.Client.CreatePullRequest(context.Background(), cfp.details.RepoOwner, cfp.details.RepoName, fixBranchName, cfp.details.Branch, commitString, prBody)
+}
+
+func (cfp *CreateFixPullRequestsCmd) createFixingBranch(impactedPackage string, fixVersionInfo *FixVersionInfo) (fixBranchName string, err error) {
+	fixBranchName, err = generateFixBranchName(cfp.details.Branch, impactedPackage, fixVersionInfo.fixVersion)
 	if err != nil {
+		return "", err
+	}
+
+	exists, err := cfp.gitManager.BranchExistsOnRemote(fixBranchName)
+	if err != nil {
+		return "", err
+	}
+	log.Info("Creating branch:", fixBranchName)
+	if exists {
+		log.Info("Branch:", fixBranchName, "already exists on remote.")
 		return
 	}
-	defer func() {
-		e := restoreDir()
-		if err == nil {
-			err = e
-		}
-	}()
 
-	// Fix all impacted packages
-	for impactedPackage, fixVersionInfo := range fixVersionsMap {
-		log.Info("-----------------------------------------------------------------")
-		log.Info("Start fixing", impactedPackage, "with", fixVersionInfo.fixVersion)
-		err = cfp.fixSinglePackageAndCreatePR(impactedPackage, *fixVersionInfo, &project, branch, repoGitParams, client, gitManager, currentWd)
-		if err != nil {
-			log.Error("failed while trying to fix and create PR for:", impactedPackage, "with version:", fixVersionInfo.fixVersion, "with error:", err.Error())
-		}
-		// After finishing to work on the current vulnerability we go back to the base branch to start the next vulnerability fix
-		log.Info("Running git checkout to base branch:", branch)
-		err = gitManager.Checkout(branch)
-		if err != nil {
-			return err
-		}
+	return fixBranchName, cfp.gitManager.CreateBranchAndCheckout(fixBranchName)
+}
 
+func (cfp *CreateFixPullRequestsCmd) cloneRepository() (tempWd string, restoreDir func() error, err error) {
+	// Create temp working directory
+	tempWd, err = fileutils.CreateTempDir()
+	if err != nil {
+		return "", nil, err
 	}
-	return nil
+	log.Debug("Created temp working directory:", tempWd)
+
+	// Clone the content of the repo to the new working directory
+	err = cfp.gitManager.Clone(tempWd, cfp.details.Branch)
+	if err != nil {
+		return tempWd, nil, err
+	}
+
+	// 'CD' into the temp working directory
+	restoreDir, err = utils.Chdir(tempWd)
+	return
 }
 
 // Create fixVersionMap - a map between impacted packages and their fix version
```

Comment:
a map between impacted packages and their fix version
-->
a map with 'impacted package' as key and 'fix version' as value.

Reply:
done